lib/IR/GraphScheduler.cpp

```diff
@@ -48,14 +48,18 @@ Scheduler *createScheduler(SchedulerKind schedulerKind, Function &G,
 
 void IRFunction::scheduleGraph(NodesPtrList &Schedule) {
   Schedule.clear();
-  for (auto &N : G_->getParent()->getConstants()) {
+  for (auto &N : G_->getConstants()) {
     Schedule.push_back(N);
   }
-  for (auto &N : G_->getParent()->getPlaceholders()) {
+  for (auto &N : G_->getPlaceholders()) {
     Schedule.push_back(N);
   }
-  auto numVars = G_->getParent()->getConstants().size();
-  auto numPlaceholders = G_->getParent()->getPlaceholders().size();
+  for (auto &N : G_->getMetadata()) {
+    Schedule.push_back(N);
+  }
+  auto numVars = G_->getConstants().size();
```

Comment:
You use `getConstants()` in this function twice, but it's no longer a simple getter -- probably should get it once at the top of the Function and reuse it to check its size. Same with `getPlaceholders()`.